src/apps/bridge/aanderaaController.cpp

```diff
@@ -0,0 +1,233 @@
+      aanderaa->direction_circ_mean_rad.addSample(degToRad(d.direction_deg_m));
+      aanderaa->direction_circ_std_rad.addSample(degToRad(d.direction_deg_m));
+      aanderaa->temp_mean_deg_c.addSample(d.temperature_deg_c);
+      if(aanderaa->abs_speed_mean_cm_s.getNumSamples() == AANDERAA_AVERAGER_MAX_SAMPLES) {
+        static char log_buf[256];
+        size_t log_buflen = snprintf(log_buf, sizeof(log_buf), "Aanderaa Node %" PRIx64 " Data: "
+                                                            "abs_speed_mean_cm_s: %.3f "
+                                                            "abs_speed_std_cm_s: %.3f "
+                                                            "direction_circ_mean_rad: %.3f "
+                                                            "direction_circ_std_rad: %.3f "
+                                                            "temp_mean_deg_c: %.3f\n",
+                                node_id, aanderaa->abs_speed_mean_cm_s.getMean(),
+                                aanderaa->abs_speed_std_cm_s.getStd(),
+                                aanderaa->direction_circ_mean_rad.getCircularMean(),
+                                aanderaa->direction_circ_std_rad.getCircularStd(),
+                                aanderaa->temp_mean_deg_c.getMean());
+        BRIDGE_LOG_PRINTN(log_buf, log_buflen);
+        // TODO transmit data to spotter once we define it. 
+        aanderaa->abs_speed_mean_cm_s.clear();
+        aanderaa->abs_speed_std_cm_s.clear();
+        aanderaa->direction_circ_mean_rad.clear();
+        aanderaa->direction_circ_std_rad.clear();
+        aanderaa->temp_mean_deg_c.clear();
+      }
+    }
+  }
+}
+
+/*!
+ * @brief Initialize the Aandera controller.
+ * This controller is responsible for detecting Aanderaa nodes and subscribing to them.
+ * It will also aggregate the data from the Aanderaa nodes and transmit it over the spotter_tx service.
+ */
+void aanderaControllerInit(BridgePowerController *power_controller) {
+  configASSERT(power_controller);
+  _ctx._bridge_power_controller = power_controller;
+  BaseType_t rval = xTaskCreate(runController, "Aandera Controller", 128 * 4, NULL,
+                                AANDERAA_CONTROLLER_TASK_PRIORITY, &_ctx._task_handle);
+  configASSERT(rval == pdTRUE);
+}
+
+static void sampleTimerCallback(TimerHandle_t timer) {
+  (void)timer;
+  xTaskNotifyGive(_ctx._task_handle);
+}
+
+static void runController(void *param) {
+  (void)param;
+  if (_ctx._initialized) {
+    configASSERT(false); // Should only be initialized once
+  }
+  _ctx._subbed_aanderaas = NULL;
+  _ctx._num_subbed_aanderaas = 0;
+  _ctx._sample_timer = xTimerCreate("AanderaaCtlTim", pdMS_TO_TICKS(SAMPLE_TIMER_MS), pdTRUE,
+                                    NULL, sampleTimerCallback);
+  configASSERT(_ctx._sample_timer);
+  configASSERT(xTimerStart(_ctx._sample_timer, 10) == pdTRUE);
+  _ctx._initialized = true;
+  while (true) {
+    ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
```

Comment:
Driven by `SAMPLE_TIMER_MS`, which is set to a 30s period. Which means the bridge won't know about the Aanderaa until 30s into boot. Not strongly attached to this number, but we want to sample rather infrequently.